### DISCUSSION TITLE
下载图片报错了

### DISCUSSION THREAD

**Discussion by HayashiKaori** [Chilfish/Weibo-archiver#15] [Q&A] [answered]
node ./download.mjs -u 2213526752
开始下载图片，请不要关闭
TypeError: fetch failed
    at node:internal/deps/undici/undici:12345:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async download_file (file:///D:/Zhuang/%E7%B4%A0%E6%9D%90/%E6%9D%82%E4%B8%83%E6%9D%82%E5%85%AB/%E5%81%B6%E5%83%8F/%E8%A6%81%E9%97%BB/preview/download.mjs:24:15)
    at async file:///D:/Zhuang/%E7%B4%A0%E6%9D%90/%E6%9D%82%E4%B8%83%E6%9D%82%E5%85%AB/%E5%81%B6%E5%83%8F/%E8%A6%81%E9%97%BB/preview/download.mjs:42:7 {
  cause: Error: getaddrinfo ENOTFOUND g2.tdimg.com
      at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:118:26) {
    errno: -3008,
    code: 'ENOTFOUND',
    syscall: 'getaddrinfo',
    hostname: 'g2.tdimg.com'
  }
}
环境：win11

**Comment by Chilfish** [comment, marked as answer]
看起来是因为这个图片的链接无法访问导致的，我忘了加这个判断😹

你可以手动修改一下 `download.mjs` 这个代码中第22行到28行的函数为下面这样。应该就会显示下载失败的原因，而不会直接退出了

```js
async function download_file(url, file_path) {
  const headers = { Referer: referer }
  try {
    const res = await fetch(url, { headers })
    const buffer = await res.arrayBuffer()
    writeFileSync(file_path, Buffer.from(buffer))
  }
  catch (e) {
    console.error(`下载失败：${url}，原因：${e.cause || e.message}`)
  }
}
```

**Reply by HayashiKaori** [reply]
好的，感谢回复。另外还有一个小问题，打开的页面，点击评论没有反应，无法看评论。

![图片](https://github.com/Chilfish/Weibo-archiver/assets/57685624/bd4c86fb-1b3d-4340-a540-8d963d6ce4a3)

**Reply by Chilfish** [reply]
是这样子的，目前的版本考虑到获取评论会频繁访问微博的接口，为了防止出现一些不可预料的来自微博方面的限制，所以暂时只获取了热评的前几条（这个你能看得到吗？）。

至于查看所有的评论，还是要点击到原微博查看（点击页面上的日期即可跳转，这个之后会加一个醒目的提示😹）

![image](https://github.com/Chilfish/Weibo-archiver/assets/60336739/932f9601-383f-46f9-a6c1-62e98109ce45)

**Reply by HayashiKaori** [reply]
现在是无法查看采集的Top15评论，点击没有反应。

**Reply by Chilfish** [reply]
好，我待会改一下，点评论按钮就跳转到原微博。请问你用的是新版本 v0.2.4 是吗？你刷新看一下我上个回复的附图，我用新版是有评论的

**Reply by HayashiKaori** [reply]
// @namespace    chilfish/monkey
// @version      0.2.4
// @author       Chilfish

**Reply by HayashiKaori** [reply]
// @namespace    chilfish/monkey
// @version      0.2.4
// @author       Chilfish

**Reply by HayashiKaori** [reply]
我的页面这样子的，换浏览器也不行：
![图片](https://github.com/Chilfish/Weibo-archiver/assets/57685624/c98f395c-ff8b-44ad-a10e-c6a84535a4e7)

**Reply by Chilfish** [reply]
emmm，你检查一下爬取的时候是否勾选了 “包括评论” 这个选项

![image](https://github.com/Chilfish/Weibo-archiver/assets/60336739/9afc11ef-cc69-439e-b0ce-d72c14bf5464)

或者按 F12，然后点击 应用，本地存储空间，weibo.com，点击右边的 fetchOptions 这个选项。我现在一时还不知道问题出在了哪里

![image](https://github.com/Chilfish/Weibo-archiver/assets/60336739/caf9da9d-27a3-45cc-a3c4-06aa449741e0)

**Reply by HayashiKaori** [reply]
是的，我这样配置。我猜测是因为爬取2015年，属于旧版接口，无法爬取评论？
如果评论也保存在文件“data.mjs”，我打开查看并无评论内容。

![图片](https://github.com/Chilfish/Weibo-archiver/assets/57685624/78831675-5a8f-41d3-85f4-1c65c7181368)

![图片](https://github.com/Chilfish/Weibo-archiver/assets/57685624/609ccd2f-0204-4c11-a024-3fb3e9003bfd)

**Comment by Chilfish** [comment]
我才注意到时间，还真可能是因为太久远的评论是没法直接获取的，要用另一个接口😹😹感谢反馈，下一版我修一下这个问题

![image](https://github.com/Chilfish/Weibo-archiver/assets/60336739/67ba06d2-9de2-4228-bb32-d7d07e5982cd)

**Comment by Chilfish** [comment]
@HayashiKaori 对了，今天下午的那笔赞赏是你发的吗？❤️先感谢支持了

因为一些限制，目前还是不支持直接点击 index.html 启动，不过再 Windows 可以写一个简单的 .bat 脚本来快速点击启动

```bat
node .\server.mjs
```

之后也会推出桌面app的，将会更方便使用和查看❤️

**Comment by Chilfish** [comment]
@HayashiKaori 我刚修复了这个 #17 bug，你试一下新版的脚本和 preview 看看，应该没问题了😇
